### PR TITLE
Assignment operators generator

### PR DESCRIPTION
**Pull request by matedabis**
It only differs from binary operators generator in the string creator method, the '=' operator,
the init function, and the expression generator.

### CODE REVIEW

**Comment by matedabis** [comment]
Updated and synchronized with #2

**Review by rerobika — COMMENTED** [review]
The code looks much cleaner. Could you show me an output?

**Comment by matedabis** [comment]
@rerobika 
```
function validate_numeric(test_value, expected_result, false_results) {
    var radix = 2 + Math.abs(expected_result % 35);
    assert(test_value === expected_result);
    assert(test_value.toString(radix) === expected_result.toString(radix));
    for (var i in false_results) {
        radix = 2 + Math.abs(false_results[i] % 35);
        assert(test_value !== false_results[i]);
        assert(test_value.toString(radix) !== false_results[i].toString(radix));
    }
}

t1 = 1045528973;
t2 = 611824280;
t3 = -1004050318;
t4 = 37311241;
t5 = 894504598;
t6 = 0;
t7 = 322296;
t8 = -1460585667;
t9 = 1300624934;
t10 = 115189529;
f1 = 718954894;
f2 = 1812465500;
f3 = 717150757;
f4 = 940592135;
f5 = -2022053499;
e = 115189529;
t1 &= t2; t1 >>= t3; t1 ^= t4; t1 |= t5; t1 <<= t6; t1 |= t7; t1 >>= t8; t1 &= t9; t1 ^= t10; 
validate_numeric(t1, e, [f1, f2, f3, f4, f5]);

...
```

File: src/scripts/generate_assignments.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        calculation = ""
+        expression = self.TEST_VALUE_VAR_NAME.format(NUMBER = 1)
+        for i in range(self.options.operand_count - 1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i + 1])
+            # If the operator is assignment, we do not need to add another
+            if result["oper"] == "=":
+                calculation += ("%s %s %s; " % (self.TEST_VALUE_VAR_NAME.format(NUMBER = 1), result["oper"],
+                self.TEST_VALUE_VAR_NAME.format(NUMBER = (i + 2))))
+            else:
+                calculation += ("%s %s= %s; " % (self.TEST_VALUE_VAR_NAME.format(NUMBER = 1), result["oper"],
```
Comment (rerobika) [inline src/scripts/generate_assignments.py]: Could we put a new line after each declaration?

File: src/scripts/generate_assignments.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        self.functions_lib(calc)
+        self.oper_funcs["assign"] = self.assign
+        self.expression_maker(calc)
+
+    # Arithmetic calculations not included in base class
+
+    # Assignment
+    def assign(self, op1, op2):
+        sum = op2
+        return {'oper': '=', 'op2': op2, 'sum': sum}
```
Comment (rerobika) [inline src/scripts/generate_assignments.py]: Why `sum` is needed?

```py
return {'oper': '=', 'op2': op2, 'sum': 'op2'}
```

Isn't this an equivalent solution?

File: src/scripts/generate_assignments.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    # Converting expression, expected value, false value in a function call to string format
+    def string_creator(self, MINNUM, MAXNUM):
+        last = self.operands[0]
+        calculation = ""
+        expression = self.TEST_VALUE_VAR_NAME.format(NUMBER = 1)
+        for i in range(self.options.operand_count - 1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i + 1])
+            # If the operator is assignment, we do not need to add another
+            if result["oper"] == "=":
+                calculation += ("%s %s %s; " % (self.TEST_VALUE_VAR_NAME.format(NUMBER = 1), result["oper"],
```
Comment (rerobika) [inline src/scripts/generate_assignments.py]: This statement is really similar to the statement below.

How about?
```py
calculation += ("%s %s %s %s\n; " % (self.TEST_VALUE_VAR_NAME.format(NUMBER = 1), result["oper"], "" if (result["oper"] == "=") else "=", self.TEST_VALUE_VAR_NAME.format(NUMBER = (i + 2))))
```

**Review by rerobika — CHANGES_REQUESTED** [review]
Getting close.

**Comment by rerobika** [comment]
The patch seems OK for me right now, but #2 must land first.

**Review by loki04 — APPROVED** [review]
LGTM after #2

**Review by rerobika — APPROVED** [review]
LGTM

**Pull request merged**